src/dcat-us/dataset-formatter.ts

```diff
@@ -1,163 +1,31 @@
-import * as _ from 'lodash';
+import { RemoteServerError } from '@esri/hub-common';
 import { adlib, TransformsList } from 'adlib';
-import { isPage } from '@esri/hub-sites';
-import { baseDatasetTemplate } from './base-dataset-template';
-import { _generateDistributions } from './_generate-distributions';
-import { cloneObject, DatasetResource, datasetToContent, getContentSiteUrls, IModel, isBBox, parseDatasetId } from '@esri/hub-common';
-import { IItem } from '@esri/arcgis-rest-portal';
-import { nonEditableFieldPaths } from './noneditable-fields';
-import { portalUrl } from '../config';
 
-// TODO - use real type for hubDataset when it gets defined in Hub.js
-type HubDatasetAttributes = Record<string, any>;
 export type DcatDatasetTemplate = Record<string, any>;
 
-export function formatDcatDataset (hubDataset: HubDatasetAttributes, siteUrl: string, siteModel: IModel, datasetTemplate: DcatDatasetTemplate) {
-  // Download and Hub Links must be generated from Content
-  const content = datasetToContent({ 
-    id: hubDataset.id, 
-    attributes: hubDataset
-  } as DatasetResource);
-  const { relative: relativePath } = getContentSiteUrls(content, siteModel);
-  const hubLandingPage = siteUrl.startsWith('https://') ? siteUrl + relativePath : `https://${siteUrl}${relativePath}`;
-  const downloadLink = siteUrl.startsWith('https://') ? `${siteUrl}/datasets/${content.identifier}` : `https://${siteUrl}/datasets/${content.identifier}`;
-
-  // AGO links must be generated from Dataset Records
-  const { itemId, layerId } = parseDatasetId(hubDataset.id);
-  let agoLandingPage = `${portalUrl}/home/item.html?id=${itemId}`;
-  if (layerId) {
-    agoLandingPage += `&sublayer=${layerId}`;
-  }
-
-  const { 
-    structuredLicense: { url = null } = {},
-    licenseInfo = ''
-  } = hubDataset;
-
-  // Override hub.js default license value of 'none'
-  if (hubDataset.license === 'none') {
-    hubDataset.license = null;
-  }
-
+export function formatDcatDataset(hubDataset: any, feedTemplate: DcatDatasetTemplate, feedTemplateTransforms: TransformsList) {
```

Comment:
This is the output plugin so it shouldn't have any knowledge that it is working with hubDatasets, so you should change the name of the param to something generic. Let's also change the name of this function to `compileDcatFeedEntry`.

Reply:
Agreed! renamed `hubDataset` to `feedData` and renamed `formatDcatDataset` to `compileDcatFeedEntry`